Contentstack.Core.Tests/AssetTest.cs

```diff
@@ -61,6 +61,24 @@ public async Task FetchAssetsPublishFallback()
             }
         }
 
+        [Fact]
+        public async Task FetchAssetsPublishMetadata()
+        {
+            List<string> list = new List<string>();
+            list.Add("en-us");
+            list.Add("ja-jp");
+            ContentstackCollection<Asset> assets = await client.AssetLibrary()
+                .SetLocale("ja-jp")
+                .IncludeMetadata()
+                .FetchAll();
+            ;
+            Assert.True(assets.Items.Count() > 0);
+            foreach (Asset asset in assets)
+            {
+                Assert.Contains((string)(asset.Get("publish_details") as JObject).GetValue("locale"), list);
```

Comment:
The assertion should verify the URL parameters. It should consider the tests passed only if the parameter "include_metadata" is included in the URL.